modules/PersistentStore/main/kvsImpl.cpp

```diff
@@ -0,0 +1,234 @@
+                                 "(@key, @value, @type)";
+    sqlite3_stmt* insert_statement;
+    sqlite3_prepare_v2(db, insert_sql_str.c_str(), insert_sql_str.size(), &insert_statement, NULL);
+
+    sqlite3_bind_text(insert_statement, 1, key.c_str(), -1, NULL);
+    sqlite3_bind_text(insert_statement, 2, string_value.c_str(), -1, NULL);
+    sqlite3_bind_text(insert_statement, 3, type.c_str(), -1, NULL);
+
+    int res = sqlite3_step(insert_statement);
+    if (res != SQLITE_DONE) {
+        EVLOG(error) << "Could not insert into KVS table: " << res << sqlite3_errmsg(db);
+        throw std::runtime_error("PersistentStore db access error");
+    }
+
+    if (sqlite3_finalize(insert_statement) != SQLITE_OK) {
+        EVLOG(error) << "Error inserting into KVS table";
+        throw std::runtime_error("PersistentStore db access error");
+    }
+};
+
+boost::variant<boost::blank, Array, Object, bool, double, int, std::string> kvsImpl::handle_load(std::string& key) {
+    std::string select_sql_str = "SELECT KEY, VALUE, TYPE FROM KVS WHERE KEY = @key";
+    sqlite3_stmt* select_statement;
+    sqlite3_prepare_v2(db, select_sql_str.c_str(), select_sql_str.size(), &select_statement, NULL);
+
+    sqlite3_bind_text(select_statement, 1, key.c_str(), -1, NULL);
+
+    int res = sqlite3_step(select_statement);
+    if (res != SQLITE_ROW) {
+        // no key with that name exists in the database
+        return {};
+    }
+
+    boost::variant<boost::blank, Array, Object, bool, double, int, std::string> value;
+
+    auto value_ptr = sqlite3_column_text(select_statement, 1);
+    if (value_ptr != nullptr) {
+        std::string value_str = std::string(reinterpret_cast<const char*>(value_ptr));
+        auto type_ptr = sqlite3_column_text(select_statement, 2);
+        if (type_ptr != nullptr) {
+            std::string type_str = std::string(reinterpret_cast<const char*>(type_ptr));
+            if (type_str == "Array") {
+                Array value_array = json::parse(value_str);
+                value = value_array;
+            } else if (type_str == "Object") {
+                Object value_object = json::parse(value_str);
+                value = value_object;
+            } else if (type_str == "bool") {
+                if (value_str == "true") {
+                    value = true;
+                } else {
+                    value = false;
+                }
+            } else if (type_str == "double") {
+                value = std::stod(value_str);
+            } else if (type_str == "int") {
+                value = std::stoi(value_str);
+            } else if (type_str == "std::string") {
+                value = value_str;
+            }
+        }
+    }
+
+    if (sqlite3_finalize(select_statement) != SQLITE_OK) {
+        EVLOG(error) << "Error selecting from KVS table";
+        throw std::runtime_error("PersistentStore db access error");
+    }
+
+    return value;
```

Comment:
concerning the interface definition we still have the problem here, that we can set a blank item, but we wont find out if it was set, because the blank return also means "not existent"